policyengine_canada/variables/gov/cra/benefits/gis_spa/gis/gis_topup_cap.py

```diff
@@ -0,0 +1,40 @@
+from policyengine_canada.model_api import *
+
+
+class gis_topup_cap(Variable):
+    value_type = float
+    entity = Person
+    label = "Guaranteed Income Supplement Top-up Maximum Value"
+    unit = CAD
+    documentation = "The highest GIS Top-up amount an individual could receive, based on the number of pensioners in the household, before the amount is reduced based on income. Roughly equivalent to imigistop in the SPSD/M variable guide."
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        gis_spa_category = person("gis_spa_category", period)
+        gis_spa_categories = gis_spa_category.possible_values
+        oas_eligible = person("oas_eligible", period)
+        p = parameters(period).gov.cra.benefits.gis_spa.topup_cap
+        return select(
```

Comment:
Could you simplify this with an `|` (or) statement, since many of the outcomes are the same? If the default isn't activated, it could be a `where`.

Same goes for the reduction